🐛 Typescript: error TS1013 raised with rest parameter in multiline function declaration

### Environment information

```bash
rome 0.6.1-next
tsc 8.3.0
```
```


### What happened?

### Summary
rome can add an invalid training comma, raising `error TS1013: A rest parameter or binding pattern may not have a trailing comma.`

### Reproduction
Given:
```typescript
// sample.ts
export class Task {
    args: any[];

    constructor(
        public script: string,
        public duration: number,
        public threadCount: number,
        ...args: any[]
    ) {
        this.args = args;
    }
}

```
`rome format --write .` will generate invalid syntax by placing a commas after `...args: any[]`

### Notes
1. The arguments in the function must be long enough to be split into multiple lines
2. The last argument must be a `rest` parameter using the `...argName` notation

### Expected result

A rest parameter in a multiline function parameter declaration will **not** have a trailing comma added.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

--- a/crates/rome_js_formatter/src/js/expressions/call_arguments.rs
+++ b/crates/rome_js_formatter/src/js/expressions/call_arguments.rs
@@ -106,7 +106,6 @@ impl FormatNodeRule<JsCallArguments> for FormatJsCallArguments {
                 // which means that if one is `false`, then the other is `true`.
                 // This means that in this branch we format the case where `should_group_first_argument`,
                 // in the else branch we format the case where `should_group_last_argument` is `true`.
-
                 write!(f, [l_leading_trivia, l_paren, l_trailing_trivia,])?;
                 if should_group_first_argument {
                     // special formatting of the first element
@@ -132,7 +131,6 @@ impl FormatNodeRule<JsCallArguments> for FormatJsCallArguments {
                         }))
                         .finish()?;
                 }
-
                 write!(f, [r_leading_trivia, r_paren, r_trailing_trivia])
             });
 

--- a/crates/rome_js_formatter/src/js/lists/constructor_parameter_list.rs
+++ b/crates/rome_js_formatter/src/js/lists/constructor_parameter_list.rs
@@ -1,5 +1,5 @@
 use crate::prelude::*;
-use rome_js_syntax::{JsConstructorParameterList, JsSyntaxKind};
+use rome_js_syntax::{JsAnyConstructorParameter, JsConstructorParameterList, JsSyntaxKind};
 
 #[derive(Debug, Clone, Default)]
 pub struct FormatJsConstructorParameterList;
@@ -8,8 +8,22 @@ impl FormatRule<JsConstructorParameterList> for FormatJsConstructorParameterList
     type Context = JsFormatContext;
 
     fn fmt(&self, node: &JsConstructorParameterList, f: &mut JsFormatter) -> FormatResult<()> {
+        // The trailing separator is disallowed if the last element in the list is a rest parameter
+        let has_trailing_rest = match node.into_iter().last() {
+            Some(elem) => matches!(elem?, JsAnyConstructorParameter::JsRestParameter(_)),
+            None => false,
+        };
+
+        let trailing_separator = if has_trailing_rest {
+            TrailingSeparator::Disallowed
+        } else {
+            TrailingSeparator::Allowed
+        };
         f.join_with(&soft_line_break_or_space())
-            .entries(node.format_separated(JsSyntaxKind::COMMA))
+            .entries(
+                node.format_separated(JsSyntaxKind::COMMA)
+                    .with_trailing_separator(trailing_separator),
+            )
             .finish()
     }
 }

--- a/crates/rome_js_formatter/src/lib.rs
+++ b/crates/rome_js_formatter/src/lib.rs
@@ -511,9 +511,18 @@ mod test {
     // use this test check if your snippet prints as you wish, without using a snapshot
     fn quick_test() {
         let src = r#"
-({aaaaaaaaaa,bbbbbbbbbb=cccccccccc,dddddddddd:eeeeeeeeee,ffffffffff:gggggggggg=hhhhhhhhhh,...jjjjjjjjjj} = x)
+export class Task {
+    args: any[];
 
-           "#;
+    constructor(
+        public script: string,
+        public duration: number,
+        public threadCount: number,
+        ...args: any[]
+    ) {
+        this.args = args;
+    }
+}       "#;
         let syntax = SourceType::ts();
         let tree = parse(src, 0, syntax);
         let result = format_node(JsFormatContext::default(), &tree.syntax())

--- a/crates/rome_js_formatter/tests/specs/js/module/class/class.js
+++ b/crates/rome_js_formatter/tests/specs/js/module/class/class.js
@@ -55,3 +55,17 @@ x = class foo extends Boar {
 
 x = class aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa extends bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb {
 }
+
+
+export class Task {
+
+
+	constructor(
+		script,
+		duration,
+		threadCount,
+		...args
+	) {
+		this.args = args;
+	}
+}

--- a/crates/rome_js_formatter/tests/specs/ts/call_expression.ts
+++ b/crates/rome_js_formatter/tests/specs/ts/call_expression.ts
@@ -8,3 +8,17 @@ export class Thing implements OtherThing {
         (type: ObjectType): Provider<Opts> => {}
     );
 }
+
+// Issue https://github.com/rome/tools/issues/2756
+export class Task {
+    args: any[];
+
+    constructor(
+        public script: string,
+        public duration: number,
+        public threadCount: number,
+        ...args: any[]
+    ) {
+        this.args = args;
+    }
+}

